src/armature/GuidingVectors.ts

```diff
@@ -101,18 +106,18 @@ export class GuidingVectors implements CostFn {
                     field.push(x, y, z);
 
                     const closest = this.closest(vec4.fromValues(x, y, z, 1));
-                    const vector = <coord>closest.curve.bezier.derivative(<number>closest.point.t);
 
                     // Make the vector as long as step/2
-                    const length = Math.sqrt(
-                        vector.x * vector.x + vector.y * vector.y + vector.z * vector.z
-                    );
-                    vector.x *= step / 2 / length;
-                    vector.y *= step / 2 / length;
-                    vector.z *= step / 2 / length;
+                    closest.guidingVector[0] *= step / 2;
```

Comment:
A `vec3` doesn't have x y and z fields since it's basically just a typed array here, but using a `vec3` means being able to use all the other vector math gl-matrix provides instead of normalizing by hand like I was doing before.